### PR TITLE
optimized Uni V3 claim fee

### PR DESCRIPTION
**Pull request by TechnoGeek01**
For getting unclaimed accrued fees of a position to display in front-end we can do a static call on nft position manager's collect function with the required tokenId, amount0max and amount1max as uint128(max) (2^128 - 1).

Increase deposit or decrease deposit won't affect our new fee collection logic. reviewed it.

### CODE REVIEW

File: test/e721-farms/uniswapv3/UniV3Farm.t.sol [hunk 1]
```diff
@@ -602,10 +602,9 @@ abstract contract ClaimUniswapFeeTest is UniV3FarmTest {
         uint256 depositId = 1;
         _simulateSwap();
         uint256 _tokenId = UniV3Farm(lockupFarm).depositToTokenId(depositId);
-        (uint256 amount0, uint256 amount1) = UniV3Farm(lockupFarm).computeUniswapFee(_tokenId);
 
-        vm.expectEmit(address(lockupFarm));
-        emit PoolFeeCollected(currentActor, _tokenId, amount0, amount1);
+        vm.expectEmit(true, false, false, false, address(lockupFarm));
```
Comment (parv3213) [inline test/e721-farms/uniswapv3/UniV3Farm.t.sol]: Let's assert all these params. We can use this 
```
(uint256 amt0, uint256 amt1) = IUniswapV3Utils(uniswapUtils).fees(pm, tokenId);
```

**Review by parv3213 — APPROVED** [review]

File: test/e721-farms/uniswapv3/UniV3Farm.t.sol (hunk 1 above)
Comment (TechnoGeek01) [reply test/e721-farms/uniswapv3/UniV3Farm.t.sol]: Yeah, since we reintroduced uniswap utils we can do this assert.

Comment (TechnoGeek01) [reply test/e721-farms/uniswapv3/UniV3Farm.t.sol]: Fixed in [76fa3da](https://github.com/Sperax/Demeter-Protocol/pull/54/commits/76fa3dadc3d37cdc08dd40be7ea3aba336aefb66)

File: test/e721-farms/uniswapv3/UniV3Farm.t.sol [hunk 2]
```diff
@@ -24,6 +24,7 @@ import {E721FarmTest, E721FarmInheritTest} from "../E721Farm.t.sol";
 import {UniV3FarmDeployer} from "../../../contracts/e721-farms/uniswapV3/UniV3FarmDeployer.sol";
 import "../../features/ExpirableFarm.t.sol";
 import "../../utils/UpgradeUtil.t.sol";
+import "../../../contracts/e721-farms/uniswapV3/interfaces/IUniswapV3Utils.sol";
```
Comment (arcantheon) [inline test/e721-farms/uniswapv3/UniV3Farm.t.sol]: Use named imports on lines 25-27

File: test/e721-farms/uniswapv3/UniV3Farm.t.sol [hunk 3]
```diff
@@ -602,10 +603,11 @@ abstract contract ClaimUniswapFeeTest is UniV3FarmTest {
...
         uint256 _tokenId = UniV3Farm(lockupFarm).depositToTokenId(depositId);
-        (uint256 amount0, uint256 amount1) = UniV3Farm(lockupFarm).computeUniswapFee(_tokenId);
+
+        (uint256 amt0, uint256 amt1) = IUniswapV3Utils(UNISWAP_UTILS).fees(NFPM, _tokenId);
 
         vm.expectEmit(address(lockupFarm));
-        emit PoolFeeCollected(currentActor, _tokenId, amount0, amount1);
+        emit PoolFeeCollected(currentActor, _tokenId, amt0, amt1);
 
         UniV3Farm(lockupFarm).claimUniswapFee(depositId);
```
Comment (arcantheon) [inline test/e721-farms/uniswapv3/UniV3Farm.t.sol:620]: Assertions missing

**Review by arcantheon — CHANGES_REQUESTED** [review]
2 comments added

File: test/e721-farms/uniswapv3/UniV3Farm.t.sol (hunk 2 above)
Comment (parv3213) [reply test/e721-farms/uniswapv3/UniV3Farm.t.sol]: Should be fine in testcases.

File: test/e721-farms/uniswapv3/UniV3Farm.t.sol (hunk 3 above)
Comment (TechnoGeek01) [reply test/e721-farms/uniswapv3/UniV3Farm.t.sol:620]: fixed in [7c12912](https://github.com/Sperax/Demeter-Protocol/pull/54/commits/7c129125969ea553e5f1584b72a7aa411c3b4179)

File: test/e721-farms/uniswapv3/UniV3Farm.t.sol (hunk 2 above)
Comment (TechnoGeek01) [reply test/e721-farms/uniswapv3/UniV3Farm.t.sol]: fixed in [1c1620a](https://github.com/Sperax/Demeter-Protocol/pull/54/commits/1c1620a1d2cc4f064aa86cc4ac93abbb2ec8d51d)

**Review by YashP16 — APPROVED** [review]

**Review by arcantheon — APPROVED** [review]

**Pull request merged**